File: algorithm/src/test/java/programmers/level0/연속된_수의_합/SolutionTest.java
package programmers.level0.연속된_수의_합;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SolutionTest {

    private Solution solution;

    @BeforeEach
    void setUp() {
        solution = new Solution();
    }

    @Test
    void test1() {
        int[] answers = solution.solution(3, 12);
        assertThat(answers).containsExactly(new int[]{3, 4, 5});
    }

    @Test
    void test2() {
        int[] answers = solution.solution(5, 15);
        assertThat(answers).containsExactly(new int[]{1, 2, 3, 4, 5});
    }

    @Test
    void test3() {
        int[] answers = solution.solution(4, 14);
        assertThat(answers).containsExactly(new int[]{2, 3, 4, 5});
    }

    @Test
    void test4() {
        int[] answers = solution.solution(5, 5);
        assertThat(answers).containsExactly(new int[]{-1, 0, 1, 2, 3});
    }

    @Test
    void test5() {
        int[] answers = solution.solution(3, 0);
        assertThat(answers).containsExactly(new int[]{-1, 0, 1});
    }
}

File: algorithm/src/test/kotlin/io/yomni/programmers/level1/id12932/SolutionTest.kt
package io.yomni.programmers.level1.id12932

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

@DisplayName("SolutionTest")
class SolutionTest {

    companion object {
        @JvmStatic
        fun argumentsProvider() = listOf(
            Arguments.of(12345, intArrayOf(5, 4, 3, 2, 1)),
            Arguments.of(98765, intArrayOf(5, 6, 7, 8, 9)),
            Arguments.of(10000, intArrayOf(0, 0, 0, 0, 1)),
        )
    }

    @ParameterizedTest
    @MethodSource("argumentsProvider")
    fun solution(input: Long, expected: IntArray) {
        val solution = Solution()
        assertThat(solution.solutionSpeed(input)).isEqualTo(expected)
    }
}
